plugins/org.eclipse.fordiac.ide.model/src/org/eclipse/fordiac/ide/model/dataimport/FBTImporter.java

```diff
@@ -427,7 +433,38 @@ private boolean handleBasicFBChildren(final BasicFBType type, final String name)
 	 * @throws XMLStreamException
 	 */
 	private void parseSimpleFB(final SimpleFBType type) throws TypeImportException, XMLStreamException {
-		processChildren(LibraryElementTags.SIMPLE_F_B_ELEMENT, name -> handleBaseFBChildren(type, name));
+		processChildren(LibraryElementTags.SIMPLE_F_B_ELEMENT, name -> handleSimpleFBChildren(type, name));
+		// create SimpleECSTate Elements if needed
+		// (this implicitly converts old type files)
+		final var inEvents = type.getInterfaceList().getEventInputs();
+		if (inEvents.size() > type.getSimpleECStates().size()) {
+			final var outEvents = type.getInterfaceList().getEventOutputs();
+			final Event stdOutEvent = !outEvents.isEmpty() ? outEvents.get(0) : null;
+			final var inEventSet = new HashSet<>(inEvents);
```

Comment:
I suggest using a `LinkedHashSet` to ensure the order is deterministic.

Reply:
Changed